raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java

```diff
@@ -0,0 +1,510 @@
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseWithAuth(PIN), generateRandomString());
+        verify(view).showOTPLayout(anyString(), anyString());
+
+    }
+
+    @Test
+    public void chargeCardWithSuggestedAuthModel_onSuccess_onAVS_VBVSECURECODESuggested_showAVS_VBVSECURECODE() {
+
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseWithAuth(AVS_VBVSECURECODE), generateRandomString());
+        verify(view).onAVSVBVSecureCodeModelUsed(anyString(), anyString());
+
+    }
+
+
+    @Test
+    public void chargeCardWithSuggestedAuthModel_onSuccess_onAVS_VBVSECURECODESuggested_showAVS_VBVSECURECODES() {
+
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateRandomChargeResponse(), generateRandomString());
+        verify(view).onPaymentError(unknownResCodemsg);
+
+    }
+
+    @Test
+    public void chargeCardWithSuggestedAuthModel_onSuccess_onUnknownAuthModelUsed_showUnknownAuthModelUsed() {
+
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseWithAuth("unknown Auth"), generateRandomString());
+        verify(view).onPaymentError(unknownAuthmsg);
+
+    }
+
+    @Test
+    public void chargeCard_onError_unknownResponse() {
+
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateNullChargeResponse(), generateRandomString());
+        verify(view).onPaymentError(invalidChargeCode);
+
+    }
+
+    @Test
+    public void chargeCardWithOTP_onValidateSuccessful_onOTPSuggested_showOTP() {
```

Comment:
@mikelis135 this test is confusing. Why does the name say "chargeCardWithOTP" and the first line of the test say "chargeCardWithAVSModel"?